code/game/mecha/equipment/tools/tools.dm

```diff
@@ -1309,5 +1332,85 @@
 		do_after_cooldown()
 	return 1
 
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool
+	name = "\improper Exosuit-Mounted Engineering Switchtool"
+	desc = "An exosuit-mounted Engineering switchtool. (Can be attached to: Any exosuit)"
+	icon_state = "mecha_rcd"	//needtochange
+	origin_tech = Tc_MATERIALS + "=3;" + Tc_POWERSTORAGE + "=2"
+	equip_cooldown = 10
+	energy_drain = 50
+	range = MELEE|RANGED
+	var/datum/global_iterator/pr_switchtool
+	var/obj/item/weapon/switchtool/engineering/mech/S
```

Comment:
Don't use single-letter variable names like this.

Reply:
Also, is it actually a good idea to use a switchtool directly here? Just seems like it's gonna cause so much bugs.